pkg/server/api/debug/v1/service.go

```diff
@@ -144,13 +104,58 @@ func (s *Service) GetInfo(ctx context.Context, req *debug_pb.GetInfoRequest) (*d
 			EntriesCount:          entries.Entries,
 			FederatedBundlesCount: bundles.Bundles,
 			SvidChain:             svidChain,
-			Uptime:                s.uptime.Unix(),
+			Uptime:                int32(s.uptime().Seconds()),
 		}
 	}
 
 	return s.getInfoResp.resp, nil
 }
 
+func (s *Service) getCertificateChain(ctx context.Context, log logrus.FieldLogger) ([]*debug.GetInfoResponse_Cert, error) {
+	trustDomainID := s.td.IDString()
+
+	// Extract trustdomains bundle and append federated bundles
+	bundle, err := s.ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
+		TrustDomainId: trustDomainID,
+	})
+	if err != nil {
+		return nil, api.MakeErr(log, codes.Internal, "failed to fetch trust domain bundle", err)
+	}
+
+	if bundle.Bundle == nil {
+		return nil, api.MakeErr(log, codes.NotFound, "bundle for trust domain not found", nil)
+	}
+
+	// Create bundle source using rootCAs
+	var rootCAs []*x509.Certificate
+	for _, b := range bundle.Bundle.RootCas {
+		cert, err := x509.ParseCertificate(b.DerBytes)
+		if err != nil {
+			return nil, api.MakeErr(log, codes.Internal, "failed to parse bundle", err)
+		}
+		rootCAs = append(rootCAs, cert)
+	}
+	bundleSource := x509bundle.FromX509Authorities(s.td, rootCAs)
+
+	// Verify certificate to extract SVID chain
+	_, certs, err := x509svid.Verify(s.so.State().SVID, bundleSource)
+	if err != nil {
+		return nil, api.MakeErr(log, codes.Internal, "failed verification against bundle", err)
+	}
+
+	// Create SVID chain for response
+	var svidChain []*debug.GetInfoResponse_Cert
+	for _, root := range certs[0] {
+		svidChain = append(svidChain, &debug_pb.GetInfoResponse_Cert{
+			Id:        spiffeIDFromCert(root),
+			ExpiresAt: root.NotAfter.Unix(),
+			Subject:   root.Subject.String(),
+		})
+	}
```

Comment:
Variable naming is confusing:
```suggestion
	_, chains, err := x509svid.Verify(s.so.State().SVID, bundleSource)
	if err != nil {
		return nil, api.MakeErr(log, codes.Internal, "failed verification against bundle", err)
	}

	// Create SVID chain for response
	var svidChain []*debug.GetInfoResponse_Cert
	for _, cert := range chains[0] {
		svidChain = append(svidChain, &debug_pb.GetInfoResponse_Cert{
			Id:        spiffeIDFromCert(cert),
			ExpiresAt: cert.NotAfter.Unix(),
			Subject:   cert.Subject.String(),
		})
	}
```